frontend/src/features/env/EnvService.ts

```diff
@@ -1,7 +1,83 @@
-import { ClientEnvVars } from '~shared/types/core'
+import {
+  ErrorDto,
+  PublicFormViewDto,
+  switchEnvFeedbackFormBodyDto,
+} from '~shared/types'
+import { ClientEnvVars, SuccessMessageDto } from '~shared/types/core'
 
+import { transformAllIsoStringsToDate } from '~utils/date'
 import { ApiService } from '~services/ApiService'
 
+import { PUBLIC_FORMS_ENDPOINT } from '~features/public-form/PublicFormService'
+
 export const getClientEnvVars = async (): Promise<ClientEnvVars> => {
   return ApiService.get<ClientEnvVars>('/client/env').then(({ data }) => data)
 }
+
+// TODO #4279: Remove after React rollout is complete
+const createFeedbackResponsesArray = (
+  formInputs: switchEnvFeedbackFormBodyDto,
+  feedbackForm: PublicFormViewDto,
+) => {
+  const responses = []
+  for (const [key, value] of Object.entries(formInputs)) {
+    const fieldIndex = key === 'url' ? 0 : key === 'feedback' ? 1 : 2
+    const { _id, fieldType } = feedbackForm?.form.form_fields[fieldIndex] ?? {}
```

Comment:
Since `feedbackForm` can no longer be `undefined`, the conditional access is no longer needed. Additionally, since my previous suggestion of using `?? {}` was to address that `undefined`, if we know for a fact that the 3 possible indexes (`0`, `1`, `2`) are all always present, it could be dropped as well.
```suggestion
    const { _id, fieldType } = feedbackForm.form.form_fields[fieldIndex]
```